### PR TITLE
Playground!

### PR DESCRIPTION
**Pull request by sidravi1**
Reviewer: @amiraliemami 
Estimate: 30mins

---

## Ticket

Closes [AAQ-196](https://idinsight.atlassian.net/browse/AAQ-196)
Closes [AAQ-273](https://idinsight.atlassian.net/browse/AAQ-273)
Closes [AAQ-255](https://idinsight.atlassian.net/browse/AAQ-255)

## Description

Adds playground (back) to Admin App. Now using Material-UI and has:
1. Choose between endpoints
2. Update API token
3. View JSON response 
4. Snackbar for Errors

### Future Tasks (optional)

1. Add Date Time
2. Add what API end point was called

## How has this been tested?

Run app locally and test!

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [x] I have updated the automated tests (if applicable)
- [] I have updated the requirements (if applicable)
- [] I have updated the README file (if applicable)
- [x] I have updated affected documentation (if applicable)

[AAQ-196]: https://idinsight.atlassian.net/browse/AAQ-196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AAQ-273]: https://idinsight.atlassian.net/browse/AAQ-273?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AAQ-255]: https://idinsight.atlassian.net/browse/AAQ-255?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: admin_app/src/app/playground/page.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      flexDirection="column"
+      alignItems="center"
+      sx={{ height: "100vh", width: "100%", pb: 10 }}
+    >
+      <Box
+        mb={10}
+        sx={{
+          width: "100%",
+          maxWidth: "lg",
+          pb: 10,
```
Comment (amiraliemami) [inline admin_app/src/app/playground/page.tsx:183]: Can we change this and the search bar width to "md"? Looks nicer when a bit tighter. I had trouble trying to get the searchbar to get smaller though - maybe some conflicts between different size declarations

File: admin_app/src/app/playground/page.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    }
+  }, []);
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      alignItems="center"
+      sx={{ height: "100vh", width: "100%", pb: 10 }}
+    >
```
Comment (amiraliemami) [inline admin_app/src/app/playground/page.tsx:177]: Making the full page white makes it look cleaner. Reference: ChatGPT's interface.


```suggestion
    <Box
      display="flex"
      flexDirection="column"
      alignItems="center"
      bgcolor="white"
      sx={{ height: "100vh", width: "100%", pb: 10 }}
    >
```

**Review by amiraliemami — CHANGES_REQUESTED** [review]
Thanks so much! Works and looks great. Left some suggestions in-code but here are some general comments.

Main change requested:
- If an API call fails, let's still respond to the user with a standard message like "Request failed. Please contact the administrator to resolve the issue." and still show the <json> response. Will help us in debugging and the user in asking for support.

Other changes (not necessarily this PR):
- We should find a way to have chat history persist across tab changes/reloads. It's quite unintuitive how it disappears. Might be a bigger task though.
- When we use px, py, mx, my, let's use standard sizing like "sizes.baseGap" so we have less hardcoding
- It would be nice to add a text section at the top of the page explaining what the playground is for, or some context.

For later:
- Let's add an endpoint to the backend to allow the admin app to auto-fetch the bearer token? This would be expected behaviour from the user's side, and we also will need it to display it once we build the "Integration" page anyway.

File: docs/components/admin-app/playground.md [hunk 3]
```diff
@@ -3,9 +3,7 @@
 The __Playground__ allows you to test the Question Answering service using a simple
 chat window.
 
-<img src="../playground-screenshot.png" alt="playground-screenshot" style="border: 1px solid  lightgray;">
+<img src="../playground.png" alt="playground-screenshot" style="border: 1px solid  lightgray;">
```
[the remark this answers is not in the document]
Comment (sidravi1) [reply docs/components/admin-app/playground.md]: Sure but them no border. I may need to create a new css class for it.

File: admin_app/src/components/PlaygroundComponents.tsx [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+  Typography,
+} from "@mui/material";
+
+import { appColors, sizes } from "@/utils";
+import CloseIcon from "@mui/icons-material/Close";
+import KeyIcon from "@mui/icons-material/Key";
+import PersonIcon from "@mui/icons-material/Person";
+import SendIcon from "@mui/icons-material/Send";
+import SmartToyIcon from "@mui/icons-material/SmartToy";
+import TextField from "@mui/material/TextField";
```
[the remark this answers is not in the document]
Comment (sidravi1) [reply admin_app/src/components/PlaygroundComponents.tsx]: Done

File: admin_app/src/components/PlaygroundComponents.tsx [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+interface ResponseMessage extends BaseMessage {
+  type: "response";
+  content: ResponseSummary[] | string;
+  json: string;
+}
+
+interface MessageBoxProps {
+  message: Message;
+  index: number;
+}
```
[the remark this answers is not in the document]
Comment (sidravi1) [reply admin_app/src/components/PlaygroundComponents.tsx]: Thanks for picking that up!

File: admin_app/src/components/PlaygroundComponents.tsx [hunk 6]
```diff
@@ -0,0 +1,426 @@
...
+      event.preventDefault();
+      onSend(queryText, selectedOption);
+      setQueryText("");
+    }
+  };
+  return (
+    <Box
+      sx={{
+        position: "fixed",
+        flexGrow: 1,
```
[the remark this answers is not in the document]
Comment (sidravi1) [reply admin_app/src/components/PlaygroundComponents.tsx]: The flexGrow? Hmm... i guess not. It is meant to stretch it to take whole row but seems to be doing that without it. I'll remove it

File: docs/blog/posts/move-to-mui.md [hunk 7]
```diff
@@ -22,9 +22,9 @@ This shift is a step towards cleaner maintainable code and a commitment to provi
 | Old UI                                                                | MUI                                                                |
 | --------------------------------------------------------------------- | ------------------------------------------------------------------ |
 | Main page                                                             |
-| ![old UI main screenshot](../../../../images/mui_switch/old_main.png) | ![MUI main screenshot](../../../../images/mui_switch/mui_main.png) |
+| ![old UI main screenshot](../images/mui_switch/old_main.png) | ![MUI main screenshot](../images/mui_switch/mui_main.png) |
 | Edit page                                                             |
-| ![old UI edit screenshot](../../../../images/mui_switch/old_edit.png) | ![MUI edit screenshot](../../../../images/mui_switch/mui_edit.png) |
+| ![old UI edit screenshot](../images/mui_switch/old_edit.png) | ![MUI edit screenshot](../images/mui_switch/mui_edit.png) |
```
[the remark this answers is not in the document]
Comment (sidravi1) [reply docs/blog/posts/move-to-mui.md:27]: If git does it's magic, there should be no conflicts!

**Review by amiraliemami — APPROVED** [review]
Thanks for the change - approving to unblock

File: admin_app/src/app/playground/page.tsx (hunk 1 above)
Comment (sidravi1) [reply admin_app/src/app/playground/page.tsx:197]: lot of the bar is taken up by other components. Will leave for now ansd we can change later if peeps want

File: docs/blog/posts/move-to-mui.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (sidravi1) [reply docs/blog/posts/move-to-mui.md:27]: Can we please leave it and see what happens?

**Pull request merged**